src/Runtime/ServiceManager.php

```diff
@@ -205,7 +205,7 @@ public function discoverPsr4Generators(): array
     {
         $classes = (new RelativeNamespaceDiscovery($this->autoloader))
             ->setRelativeNamespace('Drush\Generators')
-            ->setSearchPattern('/.*Generator\.php$/')
+            ->setSearchPattern('/.*DrushGenerator\.php$/')
```

Comment:
Rename PSR4 generators to require `DrushGenerator` not just `Generator`. Matches how PSR4 commands are named,

Reply:
It somehow escaped my attention that PSR4 commands must be named `*DrushCommands`, and not just `*Commands`. Since we already require PRS4 commands and generators to be in a "Drush" namespace, I think that it is unnecessary to require the word "Drush" in the filename and class as well. Folks already find that the need to put the word "Commands" in the file/classname is a gotcha; being inconsistent about this when commands in modules don't need it seems like a second-order gotcha that we would be better off avoiding. I would strongly recommend that we keep this as just `*Generator.php`, and loosen up PSR4 commands to also just be `*Commands.php` to keep it simpler and more straightforward. Existing PRS4 commands named `*DrushCommands.php` will keep working if we do this.

If for some reason you think that the double-Drush restriction is important for some reason, then I agree we should be consistent here.  Note also, though, that we allow either plural or signular for module commands and generators, but require plural for commands and require singular for generators when PSR-4 discovered; I'd recommend also allowing both singular and plural for PSR4 commands and generators, for consistency

Reply:
I"m fine with shortening the filename requirement. I only made it because I was seeing `sut/modules/unish/woot/src/Drush/Generators/ExampleGenerator.php` get picked up by **both** psr4 discovery and module discovery. How to avoid this double discovery problem when the filenames are the same between each? I suspect this affects both commands and generators.

Reply:
I've backed out the DrushGenerators.php suffix requirement. I'm still seeing ExampleGenerator getting discovered in $module_generators (correct) and $global_generators (incorrect). But its harmless as Application just accepts one of them. 

Lets shorten the filename requirement for commands in a new issue.

Reply:
The double discovery won't happen for most modules, because module class files are autoloaded via a Drupal-provided autoloader, and therefore do not appear in the Composer PSR-4 autoloader list. This should only be an issue for our examples. I think it's good to test that the double discovery is at least innocuous, because it's possible someone might needlessly add a PRS-4 autoload section to a module's composer.json file.